File: dmn-test-cases/tck/1.1/cl3/0006-join/translator/expected/java/dmn/type/TDeptTableImpl.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinition.ftl", "tDeptTable"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public class TDeptTableImpl implements TDeptTable {
        private java.math.BigDecimal number;
        private String name;
        private String manager;

    public TDeptTableImpl() {
    }

    public TDeptTableImpl(String manager, String name, java.math.BigDecimal number) {
        this.setManager(manager);
        this.setName(name);
        this.setNumber(number);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("manager")
    public String getManager() {
        return this.manager;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("manager")
    public void setManager(String manager) {
        this.manager = manager;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("name")
    public String getName() {
        return this.name;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("name")
    public void setName(String name) {
        this.name = name;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("number")
    public java.math.BigDecimal getNumber() {
        return this.number;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("number")
    public void setNumber(java.math.BigDecimal number) {
        this.number = number;
    }

    @Override
    public boolean equals(Object o) {
        return equalTo(o);
    }

    @Override
    public int hashCode() {
        return hash();
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: dmn-core/src/test/resources/tck/cl3/expected/kotlin/dmn/0006-join/type/TDeptTableImpl.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinition.ftl", "tDeptTable"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
class TDeptTableImpl : TDeptTable {
    @get:com.fasterxml.jackson.annotation.JsonGetter("number")
    @set:com.fasterxml.jackson.annotation.JsonGetter("number")
    override var number: java.math.BigDecimal? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("name")
    @set:com.fasterxml.jackson.annotation.JsonGetter("name")
    override var name: String? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("manager")
    @set:com.fasterxml.jackson.annotation.JsonGetter("manager")
    override var manager: String? = null

    constructor() {
    }

    constructor (manager: String?, name: String?, number: java.math.BigDecimal?) {
        this.manager = manager
        this.name = name
        this.number = number
    }

    override fun equals(other: Any?): Boolean {
        return equalTo(other)
    }

    override fun hashCode(): Int {
        return hash()
    }

    @Override
    override fun toString(): String {
        return asString()
    }
}
